fundamentals/impl.tex

```diff
@@ -392,7 +392,51 @@ \subsubsection{Resources and Materials}
 
 \subsubsection{Dynamic Resource Exchange}
 
-\TODO{Matt Gidden, please fill in}
+Dynamic Resource Exchange (DRE) is the critical resource-related process on
+which \Cyclus simulations are built. It defines the resource interaction
+mechanisms and methodologies for all agents. As the \Cyclus supply-demand
+resolution framework, the DRE must be able to support arbitrary
+facilities. Supporting the general \Cyclus philosophy, facilities are treated as
+black boxes and a clear supply-demand communication framework is defined.
+
+The DRE consists of three steps: supply-demand information gathering, resource
```

Comment:
I split the paragraphs as requested. I decided to leave out the intro paragraph, since it would consist effectively of one sentence.